system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -79,11 +83,51 @@ this.TopStoriesFeed = class TopStoriesFeed {
       this.storiesLastUpdated = Date.now();
       // This is filtered so an update function can return true to retry on the next run
       this.contentUpdateQueue = this.contentUpdateQueue.filter(update => update());
+
+      this.setCache(STORIES_CACHE_KEY, this.stories);
     } catch (error) {
       Cu.reportError(`Failed to fetch content: ${error.message}`);
     }
   }
 
+  loadCachedStories() {
+    this.stories = this.getCachedStories();
+    if (this.stories && this.stories.length > 0) {
+      this.dispatchUpdateEvent(this.storiesLastUpdated, {rows: this.stories});
+      this.storiesLastUpdated = Date.now();
+    }
+  }
+
+  loadCachedTopics() {
+    const topics = this.getCachedTopics();
+    if (topics && topics.length > 0) {
+      this.dispatchUpdateEvent(this.topicsLastUpdated, {topics, read_more_endpoint: this.read_more_endpoint});
+      this.topicsLastUpdated = Date.now();
+    }
+  }
+
+  getCachedStories() {
+    return this.getCached(STORIES_CACHE_KEY);
```

Comment:
These `getCached*` wrappers don't seem really necessary. Did you have plans for this additional indirection?

Reply:
eh. not really. just how i organized it at first. but looks silly now.